### PR TITLE
fix:zino-app#309 & fea(graphql): Some changes/improvements to typenameDataIdFromObject. And it is make cache more flexible.

### PR DESCRIPTION
**Pull request by huang12zheng**
fix:zino-app#309 & fea(graphql): Some changes/improvements to typenameDataIdFromObject. And it is make cache more flexible.
#### Fixes / Enhancements

- Fixed was zino-app#309

#### Docs

you can use "_id" for cache by:
```
DataIdFromObject  typenameDataIdFromObject=containsAllKeys(['_id']);
```

or can use 'id' just like:
```
DataIdFromObject  typenameDataIdFromObject=containsAllKeys(['id']);
```

and other case:
```
DataIdFromObject  typenameDataIdFromObject=containsAllKeys(['positionUniqStr'],'-->');
```

### CODE REVIEW

**Comment by huang12zheng** [comment]
Do I checkout a branch frome beta? and not master

**Comment by mainawycliffe** [comment]
@huang12zheng i would recommend that, it ensure you are working on the latest code since most of the times master can be behind.

**Comment by mainawycliffe** [comment]
@huang12zheng  Can you resolve merge conflicts on this PR please?

**Comment by huang12zheng** [comment]
> @huang12zheng Can you resolve merge conflicts on this PR please?

OK, I would do it.

**Comment by huang12zheng** [comment]
I was so stupid:
1. Make a "push -f" from beta,which using beta of local Repositories.It should be using origin of "zino-app".
2. I command something,"git pull https://github.com/zino-app/graphql-flutter.git beta", in local beta.It should in "addSomeCommentsForInstance".

So it take a lot time,and don't resolve conflict. And I'm so sorry about the blocking.
-------------
git checkout addSomeCommentsForInstance
git pull https://github.com/zino-app/graphql-flutter.git beta

**Comment by micimize** [comment]
@huang12zheng just wanted to let you know this is on my radar - I want to do a bit of cleanup and think more about the name and API for `compositData`.

File: packages/graphql/lib/src/utilities/helpers.dart [hunk 1]
```diff
@@ -32,6 +40,14 @@ bool areDifferentVariables(
   return areDifferent;
 }
 
+/// instances
+/// ```
+/// Map<String, dynamic> target = {"onSave":"true","onUpdate":123,"onUpdate2":{"idNo":"231","type":"IDCARD"}};
```
Comment (HofmannZ) [inline packages/graphql/lib/src/utilities/helpers.dart]: We should test these cases.

File: packages/graphql/lib/src/utilities/helpers.dart [hunk 2]
```diff
@@ -73,3 +89,55 @@ Map<String, dynamic> deeplyMergeLeft(
   // prepend an empty literal for functional immutability
   return (<Map<String, dynamic>>[{}]..addAll(maps)).reduce(_recursivelyAddAll);
 }
+
+/// instance
+/// case1
+/// 
+/// ```
+/// Map<String, Object> map={"attr1":"v1","attr2":'v2','expand':123,'attr3':'v3'};
```
Comment (HofmannZ) [inline packages/graphql/lib/src/utilities/helpers.dart]: We should test these cases.

File: packages/graphql/lib/src/utilities/helpers.dart [hunk 3]
```diff
@@ -73,3 +89,55 @@ Map<String, dynamic> deeplyMergeLeft(
...
+  return (Object object) {
+    if (objectIsCanComposit(object, keys)) return datasToString(keys, object as Map<String, Object>, seperator);
+    return null;
+  };
+}
+
+/// instance
+/// 
+/// ```
+/// Map<String, Object> map={"attr1":"v1","attr2":'v2','expand':123,'attr3':'v3'};
```
Comment (HofmannZ) [inline packages/graphql/lib/src/utilities/helpers.dart]: We should test these cases.

**Comment by HofmannZ** [comment]
Please `reword` your commit messages to align with our [contributing guidlines](https://github.com/zino-app/graphql-flutter/blob/master/CONTRIBUTING.md).

**Comment by micimize** [comment]
Will add these helpers in a different PR. Want to consider it more, given that it's essentially exposing a point-free utility function.

https://github.com/zino-app/graphql-flutter/issues/309#issuecomment-497794574

**Pull request closed without merging**